### ISSUE TITLE
Adding a parameter to an *internal* constructor should not be considered a break.

### ISSUE THREAD

**Issue by weirdan** [Roave/BackwardCompatibilityCheck#723]
Change: [vimeo/psalm@`b2ccf24` (#8961)](https://github.com/vimeo/psalm/pull/8961/commits/b2ccf2487e79df50c6b43bcf69e882789befa2dc#diff-821245150566354f0ebd05f31d6ab6bb0197aa9afaa6424041586088cf937a3c)

BCC reports the following issue (the number of required arguments changed):

![image](https://user-images.githubusercontent.com/57403/208791359-76b91293-18b1-4b14-9290-6a317a876664.png)

But it's:

1. a constructor (discussed in #497)
2. an internal method
3. of a final class

So this looks like a false-positive to me.

**Comment by weirdan** [comment]
This was reported with BCC 6.1, but the issue remains in 8.2 as well.

**Comment by Ocramius** [comment]
I'm confused by my own milestones 🤣

Taking a look shortly BTW: I think this may be a set-up problem in `bin/`

**Comment by Ocramius** [comment]
So, I tried it on `vimeo/psalm` with the patch you were referring to:

```sh
../../roave/BackwardCompatibilityCheck/bin/roave-backward-compatibility-check --from=origin/master --to=origin/pr/8961
<SNIP>
[BC] CHANGED: The number of required arguments for Psalm\Plugin\EventHandler\Event\StringInterpreterEvent#__construct() increased from 1 to 2
[BC] CHANGED: The number of required arguments for Psalm\Plugin\EventHandler\Event\BeforeAddIssueEvent#__construct() increased from 2 to 3
2 backwards-incompatible changes detected
```

I applied a config change:

```diff
diff --git a/bin/roave-backward-compatibility-check.php b/bin/roave-backward-compatibility-check.php
index 2edcf53..c261d6a 100644
--- a/bin/roave-backward-compatibility-check.php
+++ b/bin/roave-backward-compatibility-check.php
@@ -208,7 +208,7 @@ use function file_exists;
                                 )),
                             )),
                             new ClassBased\SkipClassBasedErrors(new ClassBased\MethodChanged(
-                                new MethodBased\SkipMethodBasedErrors(new MethodBased\OnlyPublicMethodChanged(
+                                new MethodBased\SkipMethodBasedErrors(new MethodBased\OnlyPublicMethodChanged(new MethodBased\ExcludeInternalMethod(
                                     new MethodBased\MultipleChecksOnAMethod(
                                         new MethodBased\SkipMethodBasedErrors(new MethodBased\MethodBecameFinal()),
                                         new MethodBased\SkipMethodBasedErrors(new MethodBased\MethodConcretenessChanged()),
@@ -226,7 +226,7 @@ use function file_exists;
                                             ),
                                         )),
                                     ),
-                                )),
+                                ))),
                             )),
                         ),
                     )),
```

```sh
../../roave/BackwardCompatibilityCheck/bin/roave-backward-compatibility-check --from=origin/master --to=origin/pr/8961
<SNIP>
No backwards-incompatible changes detected
```

Will commit change shortly :)

**Comment by Ocramius** [comment]
Handled in #724

**Comment by Ocramius** [comment]
Released BTW :)
